common-app-lib/src/cmr/common_app/api/launchpad_token_validation.clj

```diff
@@ -1,17 +1,66 @@
 (ns cmr.common-app.api.launchpad-token-validation
   "Validate Launchpad token."
   (:require
+   [cheshire.core :as json]
+   [clojure.data.codec.base64 :as base64]
+   [clojure.string :as string]
    [cmr.common-app.config :as config]
    [cmr.common.services.errors :as errors]
    [cmr.transmit.config :as transmit-config]))
 
+;; Note: Similar code exists at gov.nasa.echo.kernel.service.authentication
 (def URS_TOKEN_MAX_LENGTH 100)
 
+(defn is-heritage-token?
+  "Heritage token, meaning not the original style legacy token, but a modern
+   token made to be identifiable but also comply with the legacy standard. These
+   tokens are prefixed with 'EDL+'.
+   Note: Similar code exists at gov.nasa.echo.kernel.service.authentication."
+  [token]
+  (string/starts-with? token "EDL+"))
+
+(defn is-legacy-token?
+  "A check of last resort, only used if all other checks have not passed.
+   Note: Similar code exists at gov.nasa.echo.kernel.service.authentication."
+  [token]
+  (and (not (is-heritage-token? token))
+       (<= (count token) URS_TOKEN_MAX_LENGTH)))
+
+(defn is-jwt-token?
+  "Check if a token matches the JWT pattern (Base64.Base64.Base64) and if it
+   does, try to look inside the header section and verify that the token is JWT
+   and it came from EarthDataLogin (EDL).
+   Note: Similar code exists at gov.nasa.echo.kernel.service.authentication."
+  [token]
+  (if (some? (re-find #"[A-Za-z0-9+/=_-]+\.[A-Za-z0-9+/=_-]+\.[A-Za-z0-9+/=_-]+" token))
```

Comment:
Just a suggestion here - since it's so cheap to determine if a token is `heritage`, you could include a check in the beginning here - this could save you an expensive regex call. But, I don't know fully how this code will be used - if you already check for heritage type before ever checking for JWT type, then this is unnecessary.

Reply:
well, I don't know, I like that the function documents exactly what it is that makes a JWT token a JWT token, and you need to do the pattern check. I also want these functions to be reusable as standalone checks. Also I was asked to do it last because you may be able to get out of the code faster with the easy checks done first

Reply:
I'm ok to keep as-is. Keeping these as standalone checks definitely has value, and is probably preferable as long as the check flow is done correctly.